src/components/TouchableWithoutFocus.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {TouchableOpacity} from 'react-native';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+    /** Element that should be clickable  */
+    children: PropTypes.oneOfType([
+        PropTypes.node,
+        PropTypes.func,
+    ]).isRequired,
+
+    /** Callback for onPress event */
+    onPress: PropTypes.func.isRequired,
+
+    /** Styles that should be passed to touchable container */
+    styles: PropTypes.arrayOf(PropTypes.object),
+};
+
+const defaultProps = {
+    styles: [],
+};
+
+/**
+ * This component prevents the tapped element from capturing focus
+ */
+class TouchableWithoutFocus extends React.Component {
+    constructor(props) {
+        super(props);
+        this.onPress = this.onPress.bind(this);
+    }
+
+    onPress() {
+        if (!this.props.onPress) {
+            return;
+        }
+        this.touchableRef.blur();
```

Comment:
Instead of using a ref for this, it would also work to access the event's target passed to this method. It seems a little bit more simple.

Reply:
+1

Reply:
Tried this but didn't work as expected. TouchableOpacity gets rendered as wrapper div around the underlying image element. And focus is set on this element. 

On web click event is registered for the img element underneath the wrapper. `event.nativeEvent.currentTarget` points to app root and `event.target` point to the image element itself. So without some node querying it is not possible to locate the wrapper div which has the focus. This approach also makes so solution web centric and incompatible with native components.

Reply:
Got it. Thanks for looking into it!